src/storage/src/decode/mod.rs

```diff
@@ -582,37 +582,46 @@ where
                                 headers.as_deref(),
                             );
 
-                            let mut output = output.activate();
                             if value_bytes_remaining.is_empty() {
-                                output.session(&cap).give(DecodeResult {
-                                    key: None,
-                                    value: Some(value.map(|r| (r, 1)).map_err(|inner| {
-                                        DecodeError {
-                                            kind: inner,
-                                            raw: None,
-                                        }
-                                    })),
-                                    position: position.into(),
-                                    upstream_time_millis,
-                                    partition: partition.clone(),
-                                    metadata,
-                                });
+                                output
+                                    .give(
+                                        &cap,
+                                        DecodeResult {
+                                            key: None,
+                                            value: Some(value.map(|r| (r, 1)).map_err(|inner| {
+                                                DecodeError {
+                                                    kind: inner,
+                                                    raw: None,
+                                                }
+                                            })),
+                                            position: position.into(),
+                                            upstream_time_millis,
+                                            partition: partition.clone(),
+                                            metadata,
+                                        },
```

Comment:
Nit: extract the `DecodeResult` into a variable so the code stays readable. Here and below.

Reply:
good point, fixed